goforit_test.go

```diff
@@ -251,9 +251,11 @@ func BenchmarkEnabled(b *testing.B) {
 				g, _ := testGoforit(10*time.Microsecond, backend.backend, enabledTickerInterval)
 				defer g.Close()
 				b.ResetTimer()
-				for i := 0; i < b.N; i++ {
-					_ = g.Enabled(context.Background(), flag.flag, nil)
-				}
+				b.RunParallel(func(pb *testing.PB) {
```

Comment:
Hmm, maybe we want both the old and new versions here? We care about both latency _and_ throughput.

Reply:
@vasi-stripe I think the old version is equivalent to running the new version with `-cpu 1` on the command line (for running a single instance at once).  I can add the original form back if you think it would be less confusing (or if there is a difference I'm not seeing here?)

Reply:
I'm not really understanding why the old version benches differently at -cpu 1 vs -cpu 2! But it does look like this improves even the single-cpu results, so that's good enough for me.

Reply:
@vasi-stripe writing here for completeness:  I think whats happening is that `-cpu` internally sets `GOMAXPROCS` for each test.  for the `-cpu 1` case, we end up with contention on that single Go `proc`/processor: a single instance of the benchmark is running on it, but so is the flags backend goroutine which is calling `g.RefreshFlags()` every `10*time.Microsecond`. The non-`RunParallel` tests used default `GOMAXPROCS`, so the single goroutine running the benchmark was almost always free to run in parallel with the backend refreshing the tags.